libass/ass_parse.c

```diff
@@ -996,6 +1010,18 @@ void process_karaoke_effects(ASS_Renderer *render_priv)
             effect_type = start->effect_type;
         if (effect_type == EF_NONE)
             continue;
+        // VSF compat: \kt carries over to the next karaoke run
+        // regardless of any non-karaoke runbreaks inbetween
+        if (last_reset && !end->reset_effect) {
+            end->reset_effect = true;
+            end->effect_skip_timing = last_reset->effect_skip_timing;
```

Comment:
Good catches! The approach you proposed + also resetting `timing` for `if (start->reset_effect)`works splendidly without overriding any glyph properties.

Reply:
> \+ also resetting `timing` for `if (start->reset_effect)`

I assume you mean `skip_timing`, but I really do think you should _not_ reset it. Test: `{\kt\k100}foo{\k110\k0}bar{\k120}baz` In my understanding, “foobar” should be active for 100 ms, then _nothing_ for 110 ms, then “baz” should become active, whereas your patch activates “baz” immediately after “foobar”. `start` has `reset_effect`, but all of `skip_timing` has inevitably accumulated _after_ `start`, so it should not be discarded.

Reply:
> I assume you mean `skip_timing`

No, I actually meant `timing`, which ..uhh.. doesn't make much sense because you never mentioned removing it. Looking [at the changes](https://github.com/libass/libass/compare/dc6e8971c48bc2d765f21af543e7252d9514869c..1442342d61c9ee069118f2143cd44237067cc95e#diff-16d65434495c272051b79e3765d92ad3a268ef12b6bd32ed3f9f4d5d2d9d2829L1021-L1024) it appears I mistakenly removed the `timing` line from the conditional when applying your suggestions and then later readded it when fixing up the breakages in the comparison.
Dropping `skip_timing = 0;` while keeping `timing = 0;` makes sense to me and I confirmed this doesn't break any prior sample and your test matches xy-VSF this way, so it's now corrected. Sorry for the confusion.